tests/repository_simulator.py

```diff
@@ -185,24 +200,19 @@ def publish_root(self):
         logger.debug("Published root v%d", self.root.version)
 
     def fetch(self, url: str) -> Iterator[bytes]:
-        if not self.root.consistent_snapshot:
-            raise NotImplementedError("non-consistent snapshot not supported")
         path = parse.urlparse(url).path
         if path.startswith("/metadata/") and path.endswith(".json"):
             ver_and_name = path[len("/metadata/") :][: -len(".json")]
-            # only consistent_snapshot supported ATM: timestamp is special case
-            if ver_and_name == "timestamp":
-                version = None
-                role = "timestamp"
-            else:
-                version, _, role = ver_and_name.partition(".")
+            role, version = self._partition_filename(ver_and_name)
```

Comment:
hmm, it seems to do the right thing, but your function is called "partition filename" and then you need to feed it something that is explicitly just a portion of a filename ...